src/jdmtool/data_card/garmin.py

```diff
@@ -19,38 +22,76 @@ class GarminFirmwareWriter(BasicUsbDevice):
     WRITE_ENDPOINT = -1
     READ_ENDPOINT = -1
 
+    def write_firmware_0x300(self) -> None:
+        import time
+        print("Writing 0x300 part 1 of 2")
+        with open(FIRMWARE_DIR / 'grmn0300-part1.dat', 'rb') as fd:
+            self.write_firmware(fd)
+        time.sleep(2)
+        print("Writing 0x300 part 2 of 2")
+        with open(FIRMWARE_DIR / 'grmn0300-part2.dat', 'rb') as fd:
+            self.write_firmware(fd)
+        time.sleep(2)
+
     def write_firmware_stage1(self) -> None:
+        print("Writing stage 1")
         with open(FIRMWARE_DIR / 'grmn0500.dat', 'rb') as fd:
             self.write_firmware(fd)
 
     def init_stage2(self) -> None:
         version = self.control_read(0xC0, 0x8A, 0x0000, 0x0000, 512)
+        print("Check if stage 2 required...")
+        # this will not catch the "old" card programmer as its FW has different build time
```

Comment:
So does this cause a problem for your device? I think it would be fine to switch the logic from
```
if version != b'Aviation Card Programmer Ver 3.02 Aug 10 2015 13:21:51\x00':
```
to
```
if version == b'Aviation Card Programmer Ver 3.05 Apr 01 2024 08:42:10\x00':
```
if that solves the problem.